client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDatabaseMetaData.java

```diff
@@ -1651,4 +1651,13 @@ private static void quoteStringLiteral(StringBuilder out, String value)
         }
         out.append('\'');
     }
+
+    private String getCatalogFallback(String catalog)
+            throws SQLException
+    {
+        if (connection.getAssumeNullCatalogMeansCurrentCatalog() && catalog == null) {
+            return connection.getCatalog();
```

Comment:
can `connection.getCatalog` also return `null`?

I don't see tests for such case. i.e. create connection without a catalog defined and then even when the property is enabled it'll behave like the current behavior of listing across catalogs.

Reply:
done